lib/ansible/plugins/action/dnf.py

```diff
@@ -41,6 +41,18 @@ def run(self, tmp=None, task_vars=None):
             facts = self._execute_module(
                 module_name="ansible.legacy.setup", module_args=dict(filter="ansible_pkg_mgr", gather_subset="!all"),
                 task_vars=task_vars)
+
+            if facts.get("failed", False):
+                result.update(
+                    {
+                        "failed": True,
+                        "stderr": facts.get("module_stderr", "").strip(),
+                        "stdout": facts.get("module_stdout", "").strip(),
+                        "msg": "Failed to fetch ansible_pkg_mgr to determine the dnf module backend."
```

Comment:
`add facts.get('msg')` Also it would be cleaner to use the AnsibleActionFail exception